### DISCUSSION TITLE
post-processing

### DISCUSSION THREAD

**Discussion by carterw** [raspberrypi/picamera2#750] [Q&A]
I'm looking at the libcamera documentation and there is a section on Post-Processing;
[url]https://www.raspberrypi.com/documentation/computers/camera_software.html#post-processing[/url]

It looks very useful but the documentation talks about all of it in the context of libcamera-apps that are invoked from the command line. For example;
libcamera-hello --post-process-file /path/to/negate.json

All well and good, but how can I incorporate that post-processing into a camera app that I write in Python? The libcamera-apps are written in C++, and the source code is all available;
[url]https://github.com/search?q=repo%3Araspberrypi%2Flibcamera-apps+motion_detect+&type=code[/url]

**Comment by davidplowman** [comment]
I'm afraid the C++ post-processing is not available in Python. In the Python world people often like to do their own post-processing using OpenCV and so forth, so this kind of infrastructure has not been provided.
